subprojects/docs/src/docs/userguide/jvm/java_plugin.adoc

```diff
@@ -503,7 +503,11 @@ To help you understand how incremental compilation works, the following provides
 
 * If you are using an annotation processor that reads resources (e.g. a configuration file), you need to declare those resources as an input of the compile task.
 * If a resource file is changed, Gradle will trigger a full recompilation.
-* Using a custom `executable` or `javaHome` deactivates some optimizations. The compile task does not use incremental build immediately after a compile error or if a Java constant changes. Use <<toolchains.adoc#toolchains, toolchains>> instead if possible.
+* Using a custom `executable` or `javaHome` deactivates some optimizations.
+The compile task does not use incremental build immediately after a compile error or if a Java constant changes.
+Use <<toolchains.adoc#toolchains, toolchains>> instead if possible.
+* Having a source structure that does not match the package names, while legal for compilation, might end up causing trouble in the toolchain.
+Even more if annotation processing and <<build_cache.adoc#build_cache,caching>> are involved.
```

Comment:
```suggestion
  Even more if annotation processing and <<build_cache.adoc#build_cache,caching>> are involved.
```
Same as above -- we should indent this if we want this sentence to render within the bullet point. Consider removing this sentence entirely, though -- I think it's redundant to call out specific features that cause issues with nonstandard source structure, when we already mention that it's a bad idea. Up to you though.

Reply:
Updated

Reply:
I kept the sentence because it was that extra context that made it fail for a user recently.